src/object/obj_enum.c

```diff
@@ -387,7 +386,8 @@ fill_rec(daos_handle_t ih, vos_iter_entry_t *key_ent, struct dss_enum_arg *arg,
 
 		d_iov_set(&iov_out, iovs[arg->sgl_idx].iov_buf +
 				       iovs[arg->sgl_idx].iov_len, data_size);
-		rc = vos_iter_copy(ih, key_ent, &iov_out);
+		D_ASSERT(arg->copy_data_cb != NULL);
+		rc = arg->copy_data_cb(ih, key_ent, &iov_out);
```

Comment:
Why don't call VOS API directly?

Reply:
object client will use this file. but object client can not use VOS api.